t/ui/27-plugin_obs_rsync_batched_status_details.t

```diff
@@ -0,0 +1,183 @@
+$driver->find_element_by_link_text('Login')->click();
+
+$driver->get('/admin/obs_rsync/');
+
+is($driver->find_element('tr#folder_BatchedProj .project')->get_text(),  'BatchedProj',   'Project name');
+is($driver->find_element('tr#folder_BatchedProj .lastsync')->get_text(), '191216_150610', 'Project last sync');
+is($driver->find_element('tr#folder_BatchedProj .lastsyncversion')->get_text(), '470.1', 'Project synced version');
+# at start no project fetches version from obs
+is($driver->find_element('tr#folder_BatchedProj .obsversion')->get_text(), '', 'Project obs version empty');
+ok(index($driver->find_element('tr#folder_BatchedProj .dirtystatuscol .dirtystatus')->get_text(), 'dirty') != -1,
+    'Project dirty status');
+
+# now request fetching version from obs
+$driver->find_element('tr#folder_BatchedProj .obsversionupdate')->click();
+my $retries = 50;
+my $obsversion;
+while ($retries > 0) {
+    $obsversion = $driver->find_element('tr#folder_BatchedProj .obsversion')->get_text();
+    last if $obsversion;
+    sleep(0.1);
+    $retries = $retries - 1;
+}
+
+is($driver->find_element('tr#folder_BatchedProj .obsversion')->get_text(), '470.1', 'BatchedProj obs version');
+
+# now we call forget_run_last() and refresh_last_run() and check once again corresponding columns
+$driver->find_element('tr#folder_BatchedProj .lastsyncforget')->click();
+$driver->accept_alert;
+
+$retries = 50;
+my $lastsync;
+while ($retries > 0) {
+    $lastsync = $driver->find_element('tr#folder_BatchedProj .lastsync')->get_text();
+    last if !$lastsync;
+    sleep(0.1);
+    $retries = $retries - 1;
+}
+
+ok(index($driver->find_element('tr#folder_BatchedProj .lastsync')->get_text(), '190703_143010') == -1,
+    'BatchedProj last sync forgotten');
+is($lastsync, '', 'BatchedProj last sync forgotten');
+
+$driver->get('/admin/obs_rsync/');
+is($driver->find_element('tr#folder_BatchedProj .lastsync')->get_text(),
+    'no data', 'BatchedProj last sync is still forgotten after refresh');
+
+$driver->find_element('tr#folder_BatchedProj .dirtystatusupdate')->click();
+# now wait until gru picks the task up
+$retries = 5;
+my $dirty_status;
+while ($retries > 0) {
+    $dirty_status = $driver->find_element('tr#folder_BatchedProj .dirtystatuscol .dirtystatus')->get_text();
+    last if index($dirty_status, 'dirty') == -1;
+    sleep(1);
+    $retries = $retries - 1;
+}
+
+ok(index($dirty_status, 'dirty') == -1,     'BatchedProj dirty status is not dirty anymore');
+ok(index($dirty_status, 'published') != -1, 'BatchedProj dirty is published');
+
+# click once again and make sure that timestamp on status changed
+$driver->find_element('tr#folder_BatchedProj .dirtystatusupdate')->click();
+$retries = 5;
+my $new_dirty_status;
+while ($retries > 0) {
+    $new_dirty_status = $driver->find_element('tr#folder_BatchedProj .dirtystatuscol .dirtystatus')->get_text();
+    last if $dirty_status ne $new_dirty_status;
+    sleep(1);
+    $retries = $retries - 1;
+}
```

Comment:
Again while loops which could be for loops. And besides this is getting really repetitive. I'd prefer you create some test helper for this.